### PR TITLE
Handle interface members in NullableWalker.AsMemberOfType

### LINKED ISSUE #29967: Missing case in NullableWalker.AsMemberOfType
When accessing a member `x.Member`, and the `Member` is implemented via an interface:
```C#

        private static Symbol AsMemberOfType(NamedTypeSymbol containingType, Symbol symbol)
        {
            if (symbol is null)
            {
                return null;
            }
            if (symbol.Kind == SymbolKind.Method)
            {
                if (((MethodSymbol)symbol).MethodKind == MethodKind.LocalFunction)
                {
                    // https://github.com/dotnet/roslyn/issues/27233 Handle type substitution for local functions.
                    return symbol;
                }
            }
            var symbolDef = symbol.OriginalDefinition;
            var symbolDefContainer = symbolDef.ContainingType;
            while (true)
            {
                if (containingType.OriginalDefinition.Equals(symbolDefContainer, TypeCompareKind.ConsiderEverything))
                {
                    if (symbolDefContainer.IsTupleType)
                    {
                        return AsMemberOfTupleType((TupleTypeSymbol)containingType, symbol);
                    }
                    return symbolDef.SymbolAsMember(containingType);
                }
                containingType = containingType.BaseTypeNoUseSiteDiagnostics;
                if ((object)containingType == null)
                {
                    break;
                }
            }
            // PROTOTYPE(NullableReferenceTypes): Handle other cases such as interfaces.
            Debug.Assert(symbolDefContainer.IsInterface);
            return symbol;
        }
```

Note there is also a Watson crash involving this method: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/788968

### PR DESCRIPTION
**Pull request by cston**
Fixes #29967

### CODE REVIEW

**Comment by cston** [comment]
@dotnet/roslyn-compiler please review.

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 1]
```diff
@@ -80794,6 +80794,24 @@ class Program
                 Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(4, 23));
         }
 
+        [Fact]
+        [WorkItem(26628, "https://github.com/dotnet/roslyn/issues/26628")]
+        [WorkItem(33394, "https://github.com/dotnet/roslyn/issues/33394")]
+        public void ImplicitConstructor_03()
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >ImplicitConstructor_03 [](start = 20, length = 22)

What does "Implicit" refer to for this scenario?

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs [hunk 2]
```diff
@@ -3354,38 +3354,70 @@ private static Symbol AsMemberOfType(NamedTypeSymbol containingType, Symbol symb
...
                 }
-                containingType = containingType.BaseTypeNoUseSiteDiagnostics;
-                if ((object)containingType == null)
+            }
+            Debug.Assert(false); // If this assert fails, add an appropriate test.
+            return symbol;
+
+            bool tryAsMemberOfSingleType(NamedTypeSymbol singleType, out Symbol result)
+            {
+                if (!singleType.OriginalDefinition.Equals(symbolDefContainer, TypeCompareKind.AllIgnoreOptions))
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs]: >TypeCompareKind.AllIgnoreOptions [](start = 78, length = 32)

Do we want to match tuple names and perhaps dynamic?

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 3]
```diff
@@ -81612,5 +81630,268 @@ static void F()
...
+interface IC<T> : IA, IB<T>
+{
+}
+class Program
+{
+    static IC<T> CreateC<T>(T t)
+    {
+        throw null!;
+    }
+    static void F<T, U>()
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >, U> [](start = 19, length = 4)

It doesn't look like ```U``` is used.

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs (hunk 1 above)
Comment (cston) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: "Implicit" refers to the static constructor, although I'll admit it's confusing.

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs (hunk 2 above)
Comment (cston) [reply src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs]: Perhaps although that doesn't seem necessary.

---
In reply to: [261332320](https://github.com/dotnet/roslyn/pull/33764#discussion_r261332320) [](ancestors = 261332320)

**Comment by sharwell** [comment]
@cston I filed #33785 for the integration test timeout and restarted it

**Pull request merged**